### PR TITLE
Makes tests pass on Windows.

### PR DESCRIPTION
**Pull request by FabienTregan**
I hade 14 non passing tests under Widnows. I did not test to see if they passed undex linux or ios.

I kind of solved 10 ones (those in configure_test.go)

The solutions may not be perfect, but I wanted some feedback before I tried to continue. There is an issue.md file which contains my notes : it may be helpfull to understand what I did, but maybe you don't want to pick it in the merge.

Also I modified the big test in configure_test.go so it runs as several subtests. It allow test tools to give individual result. Maybe this can be ported to other tests that use a list of input/expected outputs.

### CODE REVIEW

**Comment by nywilken** [comment]
> would you have time to take a look at the approach taken here? 

@kytrinyx @FabienTregan I have time to take a first pass this morning.

**Comment by nywilken** [comment]
@FabienTregan thanks for working on this. I know tests on Windows were not working and have not had the time to attack it so, thank you!

I am starting to look through your proposed changes to get a sense of the approach, before leaving feed back on the changes. 

To help break things down a litter easier could you make the following changes to your PR. 

 It looks like you merged in the nextercism branch right before you started making commits, which adds unnecessary history. Please run an interactive rebase on your branch to clean up the history. 
```
git rebase -i nextercism // should open your editor allowing you to re-write history
```

Then drop all of the commits that came from the merge ( it should look like the following)
```
d c5413fc Add nextercism branch warning to PR submissions on master                                                                     
d 3199946 Fix wrapped links                                                                                                             
d 91b7798 Uninstall probot-stale on this repository                                                                                     
d 7ccbcb5 Handle alternate binary name in build script                                                                                  
d cd6e508 bin: add freebsd and openbsd releases                                                                                         
d c2c33ad iadded suppoert for linux with ppc64                                                                                          
d cccf3ec Trigger ppc64 build in build-all script                                                                                       
pick 1b34c65 Calls Normalize() on expected result in configure_test so tests passes on windows.                                         
pick 2fe4df9 Isolates the symlinks tests and ignores them under windows.                                                                
pick c8942a1 Rune each case in configure_test.go as a separate SubTest for better output.                                               
pick 2f08524 Erases the configuration files after each configure_test so new tests don't heritate old values.                           
pick 24b654d making sure the branch has everything
```
After saving your history should be clean and can be safely pushed back to up. You will need to force push your branch. 

Please let me know if you run into any issues or need help with the rebase.

File: cmd/configure_test.go [hunk 1]
```diff
@@ -1,82 +1,100 @@
...
 import (
+	"os"
+	"runtime"
 	"testing"
 
 	"github.com/exercism/cli/config"
 	"github.com/stretchr/testify/assert"
 )
 
+type testDefinition struct {
```
Comment (nywilken) [inline cmd/configure_test.go]: The changes for breaking out the tests into sub-tests should be broken into a separate PR so that we can review it separately and merge it independently of the Windows fixes.

**Comment by FabienTregan** [comment]
@nywilken Ok, I think I've done everything. I sure learnt a lot* about git. Thank you for your suggestions.

*(which means I made so much mistakes that I may not have corrected them all :^) )

**Comment by kytrinyx** [comment]
> which means I made so much mistakes that I may not have corrected them all :^)

😂 this is how we all learn! We'll just keep working with you to get all the bits and pieces together. Thanks so much for your help.

**Comment by FabienTregan** [comment]
All tests now pass on both windows and ubuntu \o/

Some solutions are not satisfying, but if you think you can merge this (probably after some review / corrections), I'll make individual PR for the one we think justify harder work.

**Comment by kytrinyx** [comment]
This is very exciting! Thank you so much for tackling the Windows work, I don't know what we would have done without your help for the upcoming launch--probably launched with a very broken experience for Windows which is exactly what we don't want.

**Comment by FabienTregan** [comment]
It was fun doing and @nywilken and your help was so kind and helpfull. But honestly I mainly corrected tests (maybe I fixed one bug or two).

Now that the tests pass, I can finally try working on the issue I first saw - provided I manage to remember what it was :)

**Comment by nywilken** [comment]
@FabienTregan great work once again in breaking up the tests into subtests. You should be able to rebase this branch onto the nextercism branch to take in your changes. I’ll try my best to give this one a review during the week.

**Comment by kytrinyx** [comment]
@FabienTregan I have force pushed up to your branch to delete the `cmd/issue.md` from the history. In the future, I would recommend putting these discoveries into the body of the pull request, or into new issues where we can discuss.

**Comment by kytrinyx** [comment]
Oh, also: I will open issues to discuss them :)
